src/Core/src/Platform/Gtk/MauiDatePicker.cs

```diff
@@ -1,35 +1,443 @@
+			{
+				if (Date < MinimumDate)
+				{
+					Date = MinimumDate;
+				}
+				else if (Date > MaximumDate)
+				{
+					Date = MaximumDate;
+				}
+			}
 		}
+	}
+
+	public class CustomComboBox : Gtk.HBox
+	{
+		private Gtk.Entry _entry;
+		private Gtk.Button _button;
+		private Gtk.Arrow _arrow;
+		private Color _color = Colors.Black;
+
+		public CustomComboBox()
+		{
+			_entry = new Gtk.Entry();
+			_entry.CanFocus = true;
+			_entry.IsEditable = true;
+			PackStart(_entry, true, true, 0);
+
+			_button = new Gtk.Button();
+			_button.WidthRequest = 30;
+			_button.CanFocus = true;
+			_arrow = new Gtk.Arrow(Gtk.ArrowType.Down, Gtk.ShadowType.EtchedOut);
+			_button.Add(_arrow);
+			PackEnd(_button, false, false, 0);
+		}
+
+		public Gtk.Entry Entry
+		{
+			get
+			{
+				return _entry;
+			}
+		}
+
+		public Gtk.Button PopupButton
+		{
+			get
+			{
+				return _button;
+			}
+		}
+
+		public Color Color
+		{
+			get { return _color; }
+			set
+			{
+				_color = value;
+				Entry.UpdateTextColor(_color);
+			}
+		}
+
+		public void SetBackgroundColor(Gdk.Color color)
+		{
+			var redColor = new Gdk.Color(255, 0, 0);
+			var blueColor = new Gdk.Color(0, 0, 255);
+			ModifyBg(Gtk.StateType.Normal, redColor);
+			Entry.OverrideBackgroundColor(Gtk.StateFlags.Normal, Colors.Blue.ToGdkRgba());
+		}
+	}
+
+	public class MauiDatePicker : EventBox
+	{
+		readonly CustomComboBox _comboBox;
+		DateTime _currentDate;
+		string? _dateFormat;
+		DatePickerWindow? _pickerWindow;
+
+		public event EventHandler DateChanged = new EventHandler((sender, args) => { });
+		public event EventHandler GotFocus = new EventHandler((sender, args) => { });
+		public event EventHandler LostFocus = new EventHandler((sender, args) => { });
+
+		public MauiDatePicker()
+		{
+			_comboBox = new CustomComboBox();
+			Add(_comboBox);
+
+			if ((Child != null))
+				Child.ShowAll();
+			Show();
 
-#pragma warning disable 169
-		// to simulate
-		// Tapping either of the DatePicker displays invokes the platform date picker
-		Gtk.Calendar? _calendar;
-#pragma warning restore 169
+			Date = DateTime.Now;
+			Format = string.Empty;
+			MinDate = new DateTime(1900, 1, 1);
+			MaxDate = new DateTime(2199, 1, 1);
 
-		DateTime _time;
+			_comboBox.Entry.CanDefault = false;
+			_comboBox.Entry.CanFocus = false;
+			_comboBox.Entry.IsEditable = false;
+			_comboBox.Entry.SetStateFlags(StateFlags.Normal, true);
+			_comboBox.Entry.CanFocus = false;
+			_comboBox.Entry.IsEditable = false;
+			_comboBox.Entry.FocusGrabbed += new EventHandler(OnEntryFocused);
+			_comboBox.PopupButton.Clicked += new EventHandler(OnBtnShowCalendarClicked);
+		}
 
 		public DateTime Date
 		{
-			get => _time;
+			get
+			{
+				return _currentDate;
+			}
 			set
 			{
-				_time = value;
-				Text = _time.ToString();
+				_currentDate = value;
+				UpdateEntryText();
 			}
 		}
 
-		public string Format { get; set; }
+		public DateTime MinDate { get; set; }
+
+		public DateTime MaxDate { get; set; }
```

Comment:
@parhamsaremi please use private setters

Reply:
Currently, we are using the setters somewhere else, but I can fix that by defining some functions.

Reply:
nah, that would defeat the purpose